src/lucky_record/where.cr

```diff
@@ -16,47 +16,99 @@ module LuckyRecord::Where
     def operator
       "="
     end
+
+    def negated : SqlClause
+      NotEqual.new(@column, @value)
+    end
   end
 
   class NotEqual < SqlClause
     def operator
       "!="
     end
+
+    def negated : SqlClause
+      Equal.new(@column, @value)
+    end
   end
 
   class GreaterThan < SqlClause
     def operator
       ">"
     end
+
+    def negated : SqlClause
+      LessThanOrEqualTo.new(@column, @value)
+    end
   end
 
   class GreaterThanOrEqualTo < SqlClause
     def operator
       ">="
     end
+
+    def negated : SqlClause
```

Comment:
I love how you handled these. The `negated` idea is great. The only suggestion I have is to change the restricted type to `LessThan` or whatever the parent type is